edx_notifications/server/api/consumer.py

```diff
@@ -211,6 +212,25 @@ def post(self, request, msg_id):
         return Response({}, status.HTTP_200_OK)
 
 
+class MarkNotifications(AuthenticatedAPIView):
+    """
+    Mark all the user notifications as read
+    """
+
+    def post(self, request):
+        """
+        HTTP POST Handler which is used for such use-cases as 'mark as read'
+        """
+        try:
+            # this will raise an ItemNotFoundError if the user_id/msg_id combo
+            # cannot be found
+            mark_all_user_notification_as_read(request.user.id)
```

Comment:
@chrisndodge, Shoaib did add the flag to mark all as unread/read, but I asked him to remove it since I cannot think of any situation where I would want to mark _all_ my notifications as unread. Marking a selected notification as unread does make sense though.